.circleci/config.yml

```diff
@@ -91,3 +80,17 @@ workflows:
           filters:
             tags:
               only: /.*/
+
+  push_eliot_image:
+    jobs:
+      - gcp-gcr/build-and-push-image:
+          name: Build and push image for eliot
+          context: eliot-circleci-image-push
+          registry-url: us-docker.pkg.dev
+          image: eliot-prod/eliot
+          tag: ${CIRCLE_TAG:-latest}
+          filters:
+            tags:
+              only: /.*/
+            branches:
+              only: main
```

Comment:
Can you describe what this is intended to do? I'm not sure I'm following it.

Reply:
This will use circleci orb https://circleci.com/orbs/registry/orb/circleci/gcp-gcr to build and push the docker image to GCP on tags or main commits.

Reply:
I don't understand how this works. How does this get run after the build-test workflow and use the image that got built?

Reply:
Instead of this, can we add a push-image step to the end of the build-test/main job after the "build docs" step?

https://circleci.com/developer/orbs/orb/circleci/gcp-gcr#commands-push-image

Reply:
makes sense, in the past we've built the image separately for tests/pushing and sort of relied on the assumption that it would be the same for the same commit. Ill figure out a way to use the same exact image then